HIP/hip-1340.md

```diff
@@ -0,0 +1,384 @@
+
+## Rationale
+
+The main proposal of this HIP is to implement EOA Code Delegation closely mimicking the original specification defined in EIP-7702.
+
+All proposed changes are a result of analysis of how Code Delegation can best fit into the existing protocol.
+
+## User Stories
+
+- As a network user I want to be able to set Code Delegation for an EOA using Ethereum-comptabile wallets and tools, so that EVM calls to my account can result in code execution.
+- As a network user I want to be able to set Code Delegation for an EOA using native CryptoCreate or CryptoUpdate transactions.
+- As a network user I want to be able to call an EOA and execute a correct delegate Smart Contract code if Code Delegation has been set for that account.
+- As a network user I want to be able to use the existing HTS/HSS/HAS facade calls with no interface changes.
+
+## Specification
+
+EOA Code Delgation has originally been defined in EIP-7702. We propose to adopt it with minimum changes.
+
+### High-Level Summary
+
+The primary feature of EOA Code Delegation is that an EOA can be configured to point to an EVM code stored in another on-chain entity (i.e. a Smart Contract).
+When such a configuration is set up, any EVM calls targetting that account will result
+in execution of the Smart Contract's EVM code in the context of original EOA (similarly to how DELEGATECALL executes the code of Contract B in the context of Contract A).
+
+![EOA Code Delegation High-Level](../assets/hip-1340/eoa_code_delegation_high_level.png)
+
+There are two main aspects of this feature:
+
+1. Being able to configure Code Delegation for an account 
+2. Executing the Code Delegation logic when an EVM call is processed
+
+both have been described in detail in dedicated sections below.
+
+### Configuring Code Delegation For an Account
+
+This section describes how Code Delegation data for a given EOA is stored and what are the means of updating it.
+
+#### Storing The Code Delegation Data
+
+We propose to adopt the following storage format, which has been originally described in EIP-7702. 
+
+Delegation information will be represented in a form of a Delegation Indicator - a fixed prefix of `0xef0100` followed by 20-byte EVM address of the delegation target.
+
+```
+delegation_indicator = 0xef0100 || 20-byte address
+```
+
+The Delegation Indicator will be stored as the code of an EOA using the facilities that currently exist for storing the code (runtime bytecode) of contract accounts - i.e. the `ContractStateStore` interface. To be able to utilize the interface, EOA's AccountID will be directly converted to a corresponding ContractID.
+
+The main revision is that the code of an EOA will become mutable. It can be either empty or contain a Delegation Indicator. No other code is allowed.
+
+As a result, all code-related queries previously available for contract accounts should now also function for EOAs, which include:
+- ContractGetBytecodeQuery
+- eth_getCode RPC method
```

Comment:
I imagine the indicator is a Hiero thing only.
Does this mean that CNs and MNs need to remove the indicator on call or is the receiver expected to remove the prefix?
Would be good to clarify